crates/astria-conductor/src/celestia/mod.rs

```diff
@@ -169,24 +177,76 @@ impl Reader {
     #[instrument(skip_all, err)]
     async fn initialize(
         &mut self,
-    ) -> eyre::Result<(executor::Handle<StateIsInit>, tendermint::chain::Id)> {
+    ) -> eyre::Result<((), executor::Handle<StateIsInit>, tendermint::chain::Id)> {
+        let validate_celestia_chain_id = async {
+            let actual_celestia_chain_id = get_celestia_chain_id(&self.celestia_client)
+                .await
+                .wrap_err("failed to fetch Celestia chain ID")?;
+            let expected_celestia_chain_id = &self.expected_celestia_chain_id;
+            ensure!(
+                self.expected_celestia_chain_id == actual_celestia_chain_id.to_string(),
```

Comment:
Why allocate? It provides a way to get a str slice which you can check against: https://docs.rs/tendermint/latest/tendermint/chain/id/struct.Id.html#method.as_str

(celestia is actually using a different tendermint library, but that one, too, should provide `as_str`)